File: C++/Top 180 Interview questions/solutions/singleelementinsortedarray.cpp
class Solution {
public:
    int singleNonDuplicate(vector<int>& nums) {
        // 1st instance - even index
        // 2nd instance - odd index
        int low = 0;
        int high = nums.size() - 2;
        while(low <= high)
        {
            int mid = (low + high) >> 1;
            if(nums[mid]==nums[mid^1])
                low = mid + 1;
            else
                high = mid - 1;
        }
        return nums[low];
    }
};	

File: Java/SingleElementInSortedArray.java
class SingleElementInSortedArray {
    public int singleNonDuplicate(int[] nums) {
        int left = 0, right = nums.length - 1;
        while( left < right){
            int mid = (left + right) / 2;
            if( mid % 2 == 1){
                mid--;
            }
            if(nums[mid] != nums[mid+1]){
                right = mid;
            }
            else {
                left = mid + 2;
            }
        }
        return nums[left];
}
}